apollo-federation/src/query_plan/operation.rs

```diff
@@ -2873,10 +2877,18 @@ impl SelectionSet {
                 // in-place, we eagerly construct the selection that needs to be rebased on the target
                 // schema.
                 let element = OpPathElement::clone(ele);
-                let selection = Selection::from_element(
-                    element,
-                    selection_set.map(|set| SelectionSet::clone(set)),
-                )?;
+                let selection_set = selection_set.map(|selection_set| {
```

Comment:
i'm really surprised linting accepts this 😅  the indentation looks so weird. does it work like this?
```suggestion
                let selection_set = selection_set
                    .map(|selection_set| {
```